test/test_asset_report.rb

```diff
@@ -0,0 +1,66 @@
+require_relative 'test_helper'
+
+# Internal: The test for Plaid::AssetReport.
+class PlaidAssetReportTest < PlaidTest
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def test_full_flow
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+    # Create an item with the "assets" product.
+    create_item initial_products: [:assets]
+
+    # Create an asset report.
+    days_requested = 60
+    options = {
+      client_report_id: '123',
+      webhook: 'https://www.example.com',
+      user: {
+        client_user_id: '789',
+        first_name: 'Jane',
+        middle_name: 'Leah',
+        last_name: 'Doe',
+        ssn: '123-45-6789',
+        phone_number: '(555) 123-4567',
+        email: 'jane.doe@example.com'
+      }
+    }
+    response = @client.asset_report.create(
+      [access_token], days_requested, options
+    )
+    refute_empty(response.asset_report_id)
+    refute_empty(response.asset_report_token)
+
+    # Poll until report generation has finished.
+    asset_report_token = response.asset_report_token
+    retries = 20
+    loop do
+      begin
+        response = @client.asset_report.get(asset_report_token)
+        break
+      rescue Plaid::PlaidAPIError => e
+        raise e if e.error_code != 'PRODUCT_NOT_READY'
+        raise 'Timed out while waiting for asset report generation' \
+          if retries.zero?
+        sleep 1
+      end
+    end
```

Comment:
You don't decrement `retries`. I'd suggest something like this:

```ruby
response = nil

20.times do
  begin
    response = @client.asset_report.get(asset_report_token)
    break     
  rescue Plaid::PlaidAPIError => e
    raise e if e.error_code != 'PRODUCT_NOT_READY'
    sleep 1    
  end
end

assert response, 'Timed out while waiting for asset report generation'
```

Reply:
⚡️